component/src/main/java/org/wso2/extension/siddhi/execution/ml/util/MathUtil.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.extension.siddhi.execution.ml.util;
+
+/**
+ * Special mathematical functions used in the ML algorithms.
+ */
+public class MathUtil {
+    /**
+     * Dot product of two 'double' vectors.
+     * @param vector1 vector 1
+     * @param vector2 vector 2
+     * @return the dot product.
+     */
+    public static double dot(double[] vector1, double[] vector2) {
+        if (vector1.length != vector2.length) {
+            throw new IllegalArgumentException("The dimensions have to be equal!");
+        }
+
+        double sum = 0;
+        for (int i = 0; i < vector1.length; i++) {
+            sum += vector1[i] * vector2[i];
+        }
+
```

Comment:
Lets remove additional new lines where applicable

Reply:
Refactored in [#adf8385974cb46e90e4050e652c5aea7f4c4629c](https://github.com/wso2-extensions/siddhi-execution-streamingml/pull/48/commits/adf8385974cb46e90e4050e652c5aea7f4c4629c)